src/App.tsx

```diff
@@ -1,21 +1,33 @@
-import { AboutMeComponent } from './components/about/AboutMe.component.tsx';
+import { useState } from 'react';
+
+import { AppContextProvider } from '@/context/AppContext.context.tsx';
+
+import { AboutComponent } from './components/about/About.component.tsx';
 import { FooterComponent } from './components/footer/Footer.component.tsx';
 import { HeaderComponent } from './components/header/Header.component.tsx';
+import { ProductsListComponent } from './components/products-list/ProductsList.component.tsx';
 
 import './App.css';
 
 function App() {
     const FULL_NAME: string = `Roman Orlovskyi`;
     const NIK_NAME: string = `roman-orlovskyi-moc`;
 
+    const [shouldShowAboutPage, setShouldShowAboutPage] = useState(true);
+
+    const toggleShowAboutPage = (isShowAboutPage: boolean) => {
+        setShouldShowAboutPage(isShowAboutPage);
+    };
+
     return (
-        <>
-            <HeaderComponent />
+        <AppContextProvider>
+            <HeaderComponent shouldShowAboutPage={shouldShowAboutPage} toggleShowAboutPageState={toggleShowAboutPage} />
             <main className="mainContentWrapper contentWrapper">
-                <AboutMeComponent fullName={FULL_NAME} nikName={NIK_NAME} />
+                {shouldShowAboutPage && <AboutComponent fullName={FULL_NAME} nikName={NIK_NAME} />}
+                {!shouldShowAboutPage && <ProductsListComponent />}
```

Comment:
You can use here ternary operator
```suggestion
                {shouldShowAboutPage ?
                  <AboutComponent fullName={FULL_NAME} nikName={NIK_NAME} /> : 
                  <ProductsListComponent />}
```

Reply:
fixed